OutOfSchool/OutOfSchool.AuthCommon/Services/ProviderAdminChangesLogService.cs

```diff
@@ -11,50 +11,31 @@ public ProviderAdminChangesLogService(IEntityRepository<long, ProviderAdminChang
         this.providerAdminChangesLogRepository = providerAdminChangesLogRepository;
```

Comment:
(Code style, probably tech debt)
`providerAdminChangesLogRepository` is of type `IEntityRepository<...>` that let you change/delete logs, not only add or read them. Is it intended? Shouldn't it be something like `IEntityAddOnlyRepository<>` ?

Reply:
@VadymLevkovskyi I added `IEntityAddOnlyRepository<>` and moved the add/read methods from `IEntityRepositoryBase` into it.

Reply:
Thank you. 
Though `RunInTransaction()` is still vulnerable, I can find reasons to put it in either interface, so let it be in new one.

Reply:
Thank you!